// @shopify/polaris/build/cjs/components/Breadcrumbs/Breadcrumbs.js
'use strict';

var React = require('react');
var polarisIcons = require('@shopify/polaris-icons');
var focus = require('../../utilities/focus.js');
var Breadcrumbs$1 = require('./Breadcrumbs.scss.js');
var Icon = require('../Icon/Icon.js');
var Text = require('../Text/Text.js');
var UnstyledLink = require('../UnstyledLink/UnstyledLink.js');

function Breadcrumbs({
  backAction
}) {
  const {
    content
  } = backAction;
  const contentMarkup = /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("span", {
    className: Breadcrumbs$1.default.Icon
  }, /*#__PURE__*/React.createElement(Icon.Icon, {
    source: polarisIcons.ArrowLeftMinor
  })), /*#__PURE__*/React.createElement(Text.Text, {
    as: "span",
    visuallyHidden: true
  }, content));
  const breadcrumbMarkup = 'url' in backAction ? /*#__PURE__*/React.createElement(UnstyledLink.UnstyledLink, {
    key: content,
    url: backAction.url,
    className: Breadcrumbs$1.default.Breadcrumb,
    onMouseUp: focus.handleMouseUpByBlurring,
    "aria-label": backAction.accessibilityLabel
  }, contentMarkup) : /*#__PURE__*/React.createElement("button", {
    key: content,
    className: Breadcrumbs$1.default.Breadcrumb,
    onClick: backAction.onAction,
    onMouseUp: focus.handleMouseUpByBlurring,
    type: "button",
    "aria-label": backAction.accessibilityLabel
  }, contentMarkup);
  return /*#__PURE__*/React.createElement("nav", {
    role: "navigation"
  }, breadcrumbMarkup);
}

exports.Breadcrumbs = Breadcrumbs;
