// @shopify/polaris/build/cjs/components/Modal/components/Header/Header.js
'use strict';

var React = require('react');
var CloseButton = require('../CloseButton/CloseButton.js');
var hooks = require('../../../../utilities/features/hooks.js');
var Box = require('../../../Box/Box.js');
var HorizontalStack = require('../../../HorizontalStack/HorizontalStack.js');
var HorizontalGrid = require('../../../HorizontalGrid/HorizontalGrid.js');
var Text = require('../../../Text/Text.js');

function Header({
  id,
  children,
  closing,
  titleHidden,
  onClose
}) {
  const {
    polarisSummerEditions2023
  } = hooks.useFeatures();
  if (titleHidden || !children) {
    return /*#__PURE__*/React.createElement(Box.Box, {
      position: "absolute",
      insetInlineEnd: polarisSummerEditions2023 ? '1' : '0',
      zIndex: "1"
    }, /*#__PURE__*/React.createElement(HorizontalStack.HorizontalStack, {
      gap: "4",
      align: "end",
      blockAlign: "center"
    }, /*#__PURE__*/React.createElement(CloseButton.CloseButton, {
      titleHidden: titleHidden,
      onClick: onClose
    })));
  }
  return /*#__PURE__*/React.createElement(Box.Box, {
    paddingBlockStart: "4",
    paddingBlockEnd: "4",
    paddingInlineStart: "5",
    paddingInlineEnd: "5",
    borderBlockEndWidth: polarisSummerEditions2023 ? undefined : '1',
    borderColor: "border-subdued",
    background: polarisSummerEditions2023 ? 'bg-subdued' : undefined
  }, /*#__PURE__*/React.createElement(HorizontalGrid.HorizontalGrid, {
    columns: {
      xs: '1fr auto'
    },
    gap: "4"
  }, /*#__PURE__*/React.createElement(HorizontalStack.HorizontalStack, {
    gap: "4",
    blockAlign: "center"
  }, /*#__PURE__*/React.createElement(Text.Text, {
    id: id,
    as: "h2",
    variant: polarisSummerEditions2023 ? 'headingMd' : 'headingLg',
    breakWord: true
  }, children)), /*#__PURE__*/React.createElement(CloseButton.CloseButton, {
    pressed: closing,
    titleHidden: titleHidden,
    onClick: onClose
  })));
}

exports.Header = Header;
