// @shopify/polaris/build/cjs/components/Card/Card.js
'use strict';

var React = require('react');
var breakpoints = require('../../utilities/breakpoints.js');
var hooks = require('../../utilities/features/hooks.js');
var Box = require('../Box/Box.js');

const Card = ({
  children,
  background = 'bg',
  padding = {
    xs: '4',
    sm: '5'
  },
  roundedAbove
}) => {
  const breakpoints$1 = breakpoints.useBreakpoints();
  const {
    polarisSummerEditions2023
  } = hooks.useFeatures();
  const defaultBorderRadius = polarisSummerEditions2023 ? '3' : '2';
  const isDefaultPadding = typeof padding !== 'string' && padding?.xs === '4' && padding?.sm === '5' && padding.md === undefined && padding.lg === undefined && padding.xl === undefined;
  const finalPadding = isDefaultPadding && polarisSummerEditions2023 ? {
    xs: '4'
  } : padding;
  let hasBorderRadius = !roundedAbove;
  if (roundedAbove && breakpoints$1[`${roundedAbove}Up`]) {
    hasBorderRadius = true;
  }
  return /*#__PURE__*/React.createElement(Box.Box, {
    background: background,
    padding: finalPadding,
    shadow: polarisSummerEditions2023 ? 'card-sm-experimental' : 'md',
    borderRadius: hasBorderRadius ? defaultBorderRadius : undefined,
    overflowX: "hidden",
    overflowY: "hidden"
  }, children);
};

exports.Card = Card;
