// @shopify/polaris/build/cjs/components/ActionList/components/Section/Section.js
'use strict';

var React = require('react');
var Item = require('../Item/Item.js');
var hooks = require('../../../../utilities/features/hooks.js');
var Box = require('../../../Box/Box.js');
var Text = require('../../../Text/Text.js');

function Section({
  section,
  hasMultipleSections,
  isFirst,
  actionRole,
  onActionAnyItem
}) {
  const {
    polarisSummerEditions2023
  } = hooks.useFeatures();
  const handleAction = itemOnAction => {
    return () => {
      if (itemOnAction) {
        itemOnAction();
      }
      if (onActionAnyItem) {
        onActionAnyItem();
      }
    };
  };
  const actionMarkup = section.items.map(({
    content,
    helpText,
    onAction,
    ...item
  }, index) => {
    return /*#__PURE__*/React.createElement(Box.Box, Object.assign({
      as: "li",
      key: `${content}-${index}`,
      role: actionRole === 'menuitem' ? 'presentation' : undefined
    }, polarisSummerEditions2023 && index !== 0 && {
      paddingBlockStart: '1'
    }), /*#__PURE__*/React.createElement(Item.Item, Object.assign({
      content: content,
      helpText: helpText,
      role: actionRole,
      onAction: handleAction(onAction)
    }, item)));
  });
  let titleMarkup = null;
  if (section.title) {
    titleMarkup = typeof section.title === 'string' ? /*#__PURE__*/React.createElement(Box.Box, polarisSummerEditions2023 ? {
      paddingBlockStart: isFirst ? '4' : '3',
      paddingInlineStart: '5',
      paddingBlockEnd: '2',
      paddingInlineEnd: '5'
    } : {
      paddingBlockStart: '4',
      paddingInlineStart: '4',
      paddingBlockEnd: '2',
      paddingInlineEnd: '4'
    }, /*#__PURE__*/React.createElement(Text.Text, {
      as: "p",
      variant: polarisSummerEditions2023 ? 'headingSm' : 'headingXs'
    }, section.title)) : /*#__PURE__*/React.createElement(Box.Box, {
      padding: "2"
    }, section.title);
  }
  let sectionRole;
  switch (actionRole) {
    case 'option':
      sectionRole = 'presentation';
      break;
    case 'menuitem':
      sectionRole = !hasMultipleSections ? 'menu' : 'presentation';
      break;
    default:
      sectionRole = undefined;
      break;
  }
  const sectionMarkup = /*#__PURE__*/React.createElement(React.Fragment, null, titleMarkup, /*#__PURE__*/React.createElement(Box.Box, Object.assign({
    as: "ul",
    padding: "2"
  }, hasMultipleSections && {
    paddingBlockStart: '0'
  }, sectionRole && {
    role: sectionRole
  }, {
    tabIndex: !hasMultipleSections ? -1 : undefined
  }), actionMarkup));
  return hasMultipleSections ? /*#__PURE__*/React.createElement(Box.Box, Object.assign({
    as: "li",
    role: "presentation",
    borderColor: polarisSummerEditions2023 ? 'border-faint-experimental' : 'border-subdued'
  }, !isFirst && {
    borderBlockStartWidth: '1'
  }, !section.title && {
    paddingBlockStart: '2'
  }), sectionMarkup) : sectionMarkup;
}

exports.Section = Section;
