// @shopify/polaris/build/esm/components/Modal/components/Header/Header.js
import React from 'react';
import { CloseButton } from '../CloseButton/CloseButton.js';
import { useFeatures } from '../../../../utilities/features/hooks.js';
import { Box } from '../../../Box/Box.js';
import { HorizontalStack } from '../../../HorizontalStack/HorizontalStack.js';
import { HorizontalGrid } from '../../../HorizontalGrid/HorizontalGrid.js';
import { Text } from '../../../Text/Text.js';

function Header({
  id,
  children,
  closing,
  titleHidden,
  onClose
}) {
  const {
    polarisSummerEditions2023
  } = useFeatures();
  if (titleHidden || !children) {
    return /*#__PURE__*/React.createElement(Box, {
      position: "absolute",
      insetInlineEnd: polarisSummerEditions2023 ? '1' : '0',
      zIndex: "1"
    }, /*#__PURE__*/React.createElement(HorizontalStack, {
      gap: "4",
      align: "end",
      blockAlign: "center"
    }, /*#__PURE__*/React.createElement(CloseButton, {
      titleHidden: titleHidden,
      onClick: onClose
    })));
  }
  return /*#__PURE__*/React.createElement(Box, {
    paddingBlockStart: "4",
    paddingBlockEnd: "4",
    paddingInlineStart: "5",
    paddingInlineEnd: "5",
    borderBlockEndWidth: polarisSummerEditions2023 ? undefined : '1',
    borderColor: "border-subdued",
    background: polarisSummerEditions2023 ? 'bg-subdued' : undefined
  }, /*#__PURE__*/React.createElement(HorizontalGrid, {
    columns: {
      xs: '1fr auto'
    },
    gap: "4"
  }, /*#__PURE__*/React.createElement(HorizontalStack, {
    gap: "4",
    blockAlign: "center"
  }, /*#__PURE__*/React.createElement(Text, {
    id: id,
    as: "h2",
    variant: polarisSummerEditions2023 ? 'headingMd' : 'headingLg',
    breakWord: true
  }, children)), /*#__PURE__*/React.createElement(CloseButton, {
    pressed: closing,
    titleHidden: titleHidden,
    onClick: onClose
  })));
}

export { Header };
