// @shopify/polaris/build/esm/components/Card/Card.js
import React from 'react';
import { useBreakpoints } from '../../utilities/breakpoints.js';
import { useFeatures } from '../../utilities/features/hooks.js';
import { Box } from '../Box/Box.js';

const Card = ({
  children,
  background = 'bg',
  padding = {
    xs: '4',
    sm: '5'
  },
  roundedAbove
}) => {
  const breakpoints = useBreakpoints();
  const {
    polarisSummerEditions2023
  } = useFeatures();
  const defaultBorderRadius = polarisSummerEditions2023 ? '3' : '2';
  const isDefaultPadding = typeof padding !== 'string' && padding?.xs === '4' && padding?.sm === '5' && padding.md === undefined && padding.lg === undefined && padding.xl === undefined;
  const finalPadding = isDefaultPadding && polarisSummerEditions2023 ? {
    xs: '4'
  } : padding;
  let hasBorderRadius = !roundedAbove;
  if (roundedAbove && breakpoints[`${roundedAbove}Up`]) {
    hasBorderRadius = true;
  }
  return /*#__PURE__*/React.createElement(Box, {
    background: background,
    padding: finalPadding,
    shadow: polarisSummerEditions2023 ? 'card-sm-experimental' : 'md',
    borderRadius: hasBorderRadius ? defaultBorderRadius : undefined,
    overflowX: "hidden",
    overflowY: "hidden"
  }, children);
};

export { Card };
